crates/contract/tests/sandbox/vote.rs

```diff
@@ -853,3 +855,63 @@ async fn test_successful_resharing_after_cancellation_clears_cancelled_epoch_id(
 
     Ok(())
 }
+
+#[tokio::test]
+async fn vote_new_parameters_errors_if_new_participant_is_missing_valid_attestation() {
+    let (worker, contract, mut current_participant_accounts, _) = init_env_secp256k1(1).await;
+
+    let state: ProtocolContractState = contract.view("state").await.unwrap().json().unwrap();
+    let ProtocolContractState::Running(initial_running_state) = state else {
+        panic!("State is not running: {:#?}", state)
+    };
+
+    let initial_epoch_id = initial_running_state.keyset.epoch_id;
+    let existing_params = initial_running_state.parameters.clone();
+    let mut new_participants = existing_params.participants().clone();
+
+    // Add a new participant
+    let (new_account, new_account_id, new_participant_info) = {
+        let (mut new_accounts, participants) = gen_accounts(&worker, 1).await;
+        let (new_account_id, _, new_participant_info) =
+            participants.participants().first().unwrap().clone();
+        let new_account = new_accounts.pop().unwrap();
+        (new_account, new_account_id, new_participant_info)
+    };
+
+    new_participants
+        .insert(new_account_id.clone(), new_participant_info)
+        .unwrap();
+
+    let threshold_parameters =
+        ThresholdParameters::new(new_participants, Threshold::new(3)).unwrap();
+
+    current_participant_accounts.push(new_account.clone());
+
+    // Vote to transition to resharing state
+    for account in &current_participant_accounts {
+        let call_result = account
+            .call(contract.id(), "vote_new_parameters")
+            .max_gas()
+            .args_json(json!({
+                "prospective_epoch_id": initial_epoch_id.next(),
+                "proposal": threshold_parameters,
+            }))
+            .transact()
+            .await
+            .unwrap()
+            .into_result()
+            .expect_err("calling `vote_new_parameters` must fail when one participant has invalid TEE status.");
+
+        let error_message = call_result.to_string();
+        let expected_error_message = InvalidParameters::InvalidTeeRemoteAttestation.to_string();
+        assert!(error_message.contains(&expected_error_message));
```

Comment:
Do we actually need to stringify before comparing? Comparing error strings feels fragile (though at least we’re not hardcoding the error strings here, which is good).